### PR TITLE
Add jsonpath module

### PR DESCRIPTION
**Pull request by zhan9san**
##### SUMMARY

Add a new module jsonpath. It is similar to xpath. But it just supports simple expressions.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

jsonpath
##### ADDITIONAL INFORMATION

It can modify the value in json file via jsonpath syntax

```paste below

```

### CODE REVIEW

**Comment by ptux** [comment]
@zhan9san 

Hi,I got your mail.

I've looked through this module.
But I cloud NOT  come up with an  use case.

I mean you could manage json files on file level with `copy` , `template` mod.

So, let me know why would we need this mod.

**Comment by zhan9san** [comment]
@ptux 
Thanks for your reply.

Actually, this module is similar to [xml](https://docs.ansible.com/ansible/latest/modules/xml_module.html#xml-module).

Assume this scene:
The same code should be deployed in many different ENVs. But as we know, different configuration files are in different ENVs.

Yes, we can do the similar thing with copy or template module. But as an Operation Engineer, we have to maintain a copy(or template file with place holder) of the configuration file written by Development. 

Once Development Engineers modify something in their Development repo, we have to sync this change in Deployment repo. If this change doesn't exist in Deployment repo, the old configuration will be deployed. Neither Dev nor Operation wants to see happen.

If we just use jsonpath or xpath(xml module) do this replacement, we won't worry about the old configuration file goes into effect.

So I think maintaining a copy is error prone and jsonpath is a better way.

Please let me know what you think. I would be obliged if you raise your suggestions.

**Comment by ptux** [comment]
@zhan9san 

OK, I got your needs.
I'll review the code.

File: lib/ansible/modules/files/jsonpath.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            path=dict(type='path', required=True),
+            jsonpath=dict(type='str', required=True),
+            value=dict(type='raw', required=True),
+        ),
+        supports_check_mode=False,
```
Comment (ptux) [inline lib/ansible/modules/files/jsonpath.py]: This module should support `check_mode`.

Comment (zhan9san) [reply lib/ansible/modules/files/jsonpath.py]: @ptux 
Thanks for your suggestion. I have enabled check_mode.
Would you please review it again?

**Comment by Akasurde** [comment]
Hi @zhan9san, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.general<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**